### ISSUE TITLE
Depend on libgdal-core and any needed plugins

### ISSUE THREAD

**Issue by akrherz** [conda-forge/r-sf-feedstock#137]
### Comment:

Presently, this feedstock depends on `libgdal` (the kitchen sink).  Can it be updated to depend on `libgdal-core` and any required plugins needed for functionality?

**Comment by mfansler** [comment]
I think this is a good suggestion. I'm on break for a few weeks but would be happy to review a PR. Or maybe @danielnachun might want to look into it.

**Comment by danielnachun** [comment]
This should be possible.  How do we determine which plugins are actually needed?

**Comment by akrherz** [comment]
Thanks for the responses.  I am the least sharp tool in the toolbox, but I inspect the build logs and look for the sf.so links

INFO (r-sf,lib/R/library/sf/libs/sf.so): Needed DSO lib/libgdal.so.36 found in conda-forge/linux-64::libgdal-core==3.10.0=hef9eae6_1

So unless r-sf calls command line stuff provided by the plugins, it would seem all it needs is libgdal-core, but I could be horribly wrong, sorry.

**Comment by danielnachun** [comment]
I think you're right.  I tried a local build with only `libgdal-core` and it worked with no issues.  I see no references in the [C++ source code](https://github.com/cran/sf) to `dlopen` which as far as I know is the only way you load plugins at run time that were not linked at build time.  I'm also not even sure Rcpp supports dynamic loading of shared libraries in that matter.  I've opened https://github.com/conda-forge/r-sf-feedstock/pull/138 so we'll see how it goes.

If this does work, I'll also do the same `r-terra`, as that also currently depends on the full `libgdal` but probably doesn't use plugins.
